org.opent2t.sample.lamp.superpopular/com.wink.lightbulb/js/thingTranslator.js

```diff
@@ -36,17 +36,29 @@ class StateReader {
     }
 }
 
+// Helper method to convert Wink's 0.0-1.0 brightness scale to a 0-100 scale
+function scaleBrightnessToDim(brightnessValue) {
+    return Math.floor(brightnessValue * 100);
+}
+
+// Helper method to convert a 0-100 scale to Wink's 0.0-1.0 brightness scale
+function scaleDimToBrightness(dimmingValue) {
+    return dimmingValue / 100;
+}
+
 // Helper method to convert the device schema to the translator schema.
 function deviceSchemaToTranslatorSchema(deviceSchema) {
     var stateReader = new StateReader(deviceSchema.desired_state, deviceSchema.last_reading);
 
     var powered = stateReader.get('powered');
+    var brightness = stateReader.get('brightness');
 
     return {
         id: deviceSchema['object_type'] + '.' + deviceSchema['object_id'],
```

Comment:
It was somewhat arbitrary, but for Wink, the minimum set of information to interact with a specific device is the object type and the object ID, so that's what I went with when implementing the thermostat. For the Lamp, I believe the same style carried over from that decision.

I think it would probably be good to have a deeper discussion about how we should treat this ID field.

With Wink in particular, there is a problem with trying to use just the object_id to identify a device over time, since, according to Wink's documentation:

"It is possible for the API to re-assign identifiers to resources to rebalance keys; in this case, your resource will still exist but it (and all references to it) will be updated to the new identifier. Your application should be able to handle this case."

Wink exposes a uuid field that should uniquely identify a device, regardless of its object_id, however, you cannot actually interact with the device without the object_id.

I think there is value in exposing the uuid field, since without it, I think it's impossible for an application using openT2T to handle the case where the object_id is re-assigned, so I would prefer that we use that here, but then we don't have the object_id in the device schema at all, so I don't know if that breaks any scenarios.

---
In reply to: [76939288](https://github.com/openT2T/translators/pull/87#discussion_r76939288) [](ancestors = 76939288)